zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,297 @@
+//! A type that can hold the four types of Zcash value pools.
+
+use crate::amount::{Amount, Constraint, Error, NegativeAllowed, NonNegative};
+
+/// An amount spread between different Zcash pools.
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(bound = "C: Constraint")]
+pub struct ValueBalance<C = NegativeAllowed> {
+    transparent: Amount<C>,
+    sprout: Amount<C>,
+    sapling: Amount<C>,
+    orchard: Amount<C>,
+}
+
+impl<C> ValueBalance<C>
+where
+    C: Constraint + Copy,
+{
```

Comment:
Optional: We might be missing the following methods or implementations:
* `constrain`
* `to_bytes` and `from_bytes`
* arithmetic assignment operators
* a `Sum` implementation

But if they aren't needed, or they get added in a future PR, that is also ok.

Reply:
Yes, i know `Sum` is going to be needed for sure, probably some of the others too but lets leave it for the next PR that start to use this new type.